### PR TITLE
first draft of plan for git session

### PR DESCRIPTION
**Pull request by tobyhodges**
How does this look to you, @hdinkel? 

Still needs more detail about what we will actually put into the Python script that we're making as we go along. As mentioned on our last call, I suggest something that calculates some basic statistics from a FASTA file provided, e.g. count of sequences, mean length, longest sequence.

### CODE REVIEW

File: git/git-plan.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+- what is version control?
+- why should I use it?
+- what is git?
+- what is GitHub?
+
+## GitHub
+
+- look at an example repo
+    - e.g. this one? One of the Software Carpentry lessons? any other suggestions?
```
Comment (hdinkel) [inline git/git-plan.md:13]: yeah why not this one. Has a README, has LICENSE, has decent amount of commits.

File: git/git-plan.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    - commit messages - what they are; advice for writing good ones
+    - look at history
+    - repeat steps above
+    - add a new file (python script), commit, look at result
+- discuss limitations of online-only approach
+    - how can we be sure that our script is doing what we intend?
+    - would like to make changes, test, then update the material online when we're sure that it's ready
+- now switch to the command line
+
+## Command Line
```
Comment (hdinkel) [inline git/git-plan.md:29]: We should consider handing out a git cheat sheet

File: git/git-plan.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  - `git config --global core.editor "nano -w"`
+- mention `git init`
+- `git status`
+- `git add`
+- `git commit`
+  - commit messages
+  - imagine your future self as a collaborator, who won't know (remember) why you made the changes you're making
+- `git add`
+- `git commit -m`
+- `git commit -a`
```
Comment (hdinkel) [inline git/git-plan.md]: Not a big fan of `git commit -a` as it encourages committing without thinking what to commit. IMHO staging is an important step.

**Review by hdinkel — APPROVED** [review]
Nice workflow!

File: git/git-plan.md (hunk 3 above)
Comment (tobyhodges) [reply git/git-plan.md]: Very valid point. Happy to leave this out.

File: git/git-plan.md (hunk 1 above)
Comment (tobyhodges) [reply git/git-plan.md:13]: 👍

**Comment by tobyhodges** [comment]
I'm going to merge this after I remove the point for `commit -a`. I still need to draft a small Python script for summarising FASTA files, for us to use during this lesson, but that will have to wait until after the instructor training next week.

**Pull request merged**